hagglehaul.Server/Controllers/DriverController.cs

```diff
@@ -231,28 +233,184 @@ public async Task<IActionResult> GetAllAvailableTrips()
+        )
+        {
+            if (cache != null)
+                return cache;
+            var tripRoutes = new Dictionary<string, GeographicRoute>();
+            await Task.WhenAll(trips.Select(async trip =>
+            {
+                var route = await _geographicRouteService.GetGeographicRoute(
+                    trip.PickupLong,
+                    trip.PickupLat,
+                    trip.DestinationLong,
+                    trip.DestinationLat
+                );
+                tripRoutes.Add(trip.Id, route);
+            }));
+            return tripRoutes;
+        }
+
+        private async Task<Dictionary<string, uint>> GetTripIdToMinBidAmountMap(
+            List<Trip> trips,
+            Dictionary<string, uint>? cache
+        )
+        {
+            if (cache != null)
+                return cache;
+            var tripMinBidAmounts = new Dictionary<string, uint>();
+            await Task.WhenAll(trips.Select(async trip =>
+            {
+                var bids = await _bidService.GetTripBidsAsync(trip.Id);
+                var bidAmounts = bids.Select(b => b.CentsAmount);
+                var minBidAmount = bidAmounts.Min();
+                tripMinBidAmounts.Add(trip.Id, minBidAmount);
+            }));
+            return tripMinBidAmounts;
+        }
+
+        private double EuclideanDistance(double lat1, double long1, double lat2, double long2)
+        {
+            double dLat = lat2 - lat1;
+            double dLong = long2 - long1;
+            return Math.Sqrt(dLat * dLat + dLong * dLong);
+        }
+
+        private double TripEuclideanDistance(Trip trip)
+        {
+            return EuclideanDistance(
+                trip.PickupLat,
+                trip.PickupLong,
+                trip.DestinationLat,
+                trip.DestinationLong
+            );
+        }
+
+        private double TripCurrentToStartDistance(Trip trip, TripMarketOptions options)
+        {
+            return EuclideanDistance(
+                (double)options.CurrentLat,
+                (double)options.CurrentLong,
+                trip.PickupLat,
+                trip.PickupLong
+            );
+        }
+
+        private double TripEndToTargetDistance(Trip trip, TripMarketOptions options)
+        {
+            return EuclideanDistance(
+                trip.DestinationLat,
+                trip.DestinationLong,
+                (double)options.TargetLat,
+                (double)options.TargetLong
+            );
+        }
+
+        private bool TripPassesMaxCurrentToStartDistance(Trip trip, TripMarketOptions options)
+        {
+            return TripCurrentToStartDistance(trip, options) <= options.MaxCurrentToStartDistance;
+        }
+
+        private bool TripPassesMaxEndToTargetDistance(Trip trip, TripMarketOptions options)
+        {
+            return TripEndToTargetDistance(trip, options) <= options.MaxEndToTargetDistance;
+        }
+
         [HttpGet]
         [Route("tripMarket")]
```

Comment:
On all `HttpGet` routes which return `IActionResult` you should add this annotation:

`[ProducesResponseType(typeof(your return type here), StatusCodes.Status200OK)]`